extensions/CubesterYT/KeySimulation.js

```diff
@@ -0,0 +1,194 @@
+
+  let delay = (ms) => new Promise((res) => setTimeout(res, ms));
+
+  class KeySimulation {
+    getInfo() {
+      return {
+        id: "cubesterKeySimulation",
+        name: "Key Simulation",
+        color1: "#BF0000",
+        color2: "#800000",
+        menuIconURI: icon,
+        docsURI: "https://extensions.turbowarp.org/CubesterYT/KeySimulation",
+
+        blocks: [
+          {
+            opcode: "pressKey",
+            text: "press key [KEY]",
+            blockType: Scratch.BlockType.COMMAND,
+            arguments: {
+              KEY: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "KEYS",
+              },
+            },
+          },
+          {
+            opcode: "holdKey",
+            text: "hold key [KEY] for [SECONDS] seconds",
+            blockType: Scratch.BlockType.COMMAND,
+            arguments: {
+              KEY: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "KEYS",
+              },
+              SECONDS: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: 5,
+              },
+            },
+          },
+          {
+            opcode: "clickMouse",
+            text: "click mouse at x: [X] y: [Y]",
+            blockType: Scratch.BlockType.COMMAND,
+            arguments: {
+              X: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: 0,
+              },
+              Y: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: 0,
+              },
+            },
+          },
+          {
+            opcode: "holdMouse",
+            text: "hold mouse at x: [X] y: [Y] for [SECONDS] seconds",
```

Comment:
"Click and hold" would describe this block better; as it stands, it sounds like it would just move the mouse pointer to that location for a given time.

Reply:
I haven't fully decided what I should put there, so I put something close to functionality.

Reply:
@NexusKitten I've taken up your suggestion after thinking about it, it does make sense to say click and hold, will be changing it now.